python/dgl/graph_index.py

```diff
@@ -45,28 +46,29 @@ def __setstate__(self, state):
         """The pickle state of GraphIndex is defined as a triplet
         (number_of_nodes, multigraph, readonly, src_nodes, dst_nodes)
         """
-        n_nodes, multigraph, readonly, src, dst = state
+        num_nodes, multigraph, readonly, src, dst = state
 
         self._cache = {}
         self._multigraph = multigraph
         self._readonly = readonly
-        if readonly:
-            self._init(src, dst, n_nodes)
+        if multigraph is None:
+            self._handle = _CAPI_DGLGraphCreate1(
+                src.todgltensor(),
+                dst.todgltensor(),
+                int(num_nodes),
+                readonly)
         else:
-            self._handle = _CAPI_DGLGraphCreateMutable(multigraph)
-            self.clear()
-            self.add_nodes(n_nodes)
-            self.add_edges(src, dst)
-
-    def _init(self, src_ids, dst_ids, num_nodes):
-        """The actual init function"""
-        assert len(src_ids) == len(dst_ids)
-        self._handle = _CAPI_DGLGraphCreate(
-            src_ids.todgltensor(),
-            dst_ids.todgltensor(),
-            self._multigraph,
-            int(num_nodes),
-            self._readonly)
+            self._handle = _CAPI_DGLGraphCreate(
+                src.todgltensor(),
+                dst.todgltensor(),
+                multigraph,
+                int(num_nodes),
+                readonly)
```

Comment:
will it be better to use one API and the values of multigraph can be -1, 0, 1? too hacky?

Reply:
That's fine as long as it is not exposed to user side. Will do that to eliminate some copy-pasting.

Reply:
Done